packages/service-library/src/data-providers/page-document-provider.ts

```diff
@@ -34,104 +36,26 @@ export class PageDocumentProvider {
         };
     }
 
-    public async getAllPages(websiteId: string, itemCount?: number): Promise<WebsitePage[]> {
-        const pages: WebsitePage[] = [];
-        let continuationToken: string;
-        do {
-            const response = await this.getPages(websiteId, continuationToken, itemCount);
-            client.ensureSuccessStatusCode(response);
-            pages.push(...response.item);
-            continuationToken = response.continuationToken;
-        } while (continuationToken !== undefined);
+    public async getReadyToScanPagesForWebsite(websiteId: string, itemCount: number = 1): Promise<WebsitePage[]> {
+        const pagesNotScannedBefore = await this.getPagesNotScannedBefore(websiteId, itemCount);
```

Comment:
Just out of curiosity - Can we combine these two queries (scanned pages and not scanned pages) and order by something (to keep the pages not scanned ahead of the pages scanned at least once) to make a single call instead of multiple calls?

Reply:
cosmos sort query doesnt include results that do not have this property. so, we need to use 2 queries in this case